src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -203,4 +206,31 @@ protected Object readResolve() {
         return this;
     }
 
+    public void taskAccepted(Executor executor, Queue.Task task) {
+        return;
+    }
+
+    public void taskCompleted(Executor executor, Queue.Task task, long durationMS) {
+        postJobAction(executor);
+    }
+
+    public void taskCompletedWithProblems(Executor executor, Queue.Task task, long durationMS, Throwable problems) {
+        postJobAction(executor);
+    }
+
+    private void postJobAction(Executor executor){
+        EC2Computer computer = (EC2Computer) executor.getOwner();
+        EC2AbstractSlave slaveNode = computer.getNode();
+        int maxTotalUses = Integer.parseInt(slaveNode.maxTotalUses);
```

Comment:
What should happen here if it's not a valid integer? If you use a field type of int (or Integer), Stapler should take care of conversions and error messages I think.